finance_dl/scrape_lib.py

```diff
@@ -140,42 +140,63 @@ def __init__(self, download_dir=None, connect=None, chromedriver_bin='finance-dl
-        chrome_options = webdriver.ChromeOptions()
-        service_args = []
-        chrome_options.add_experimental_option('excludeSwitches', [
-            'enable-automation',
-            'load-extension',
-            'load-component-extension',
-            'ignore-certificate-errors',
-            'test-type',
-        ])
-        if profile_dir is not None:
-            chrome_options.add_argument('user-data-dir=%s' % profile_dir)
-            if not os.path.exists(profile_dir):
-                os.makedirs(profile_dir)
-        prefs = {}
-        prefs['plugins.plugins_disabled'] = [
-            'Chrome PDF Viewer', 'Chromium PDF Viewer'
-        ]
-        prefs['plugins.always_open_pdf_externally'] = True
-        if download_dir is not None:
-            prefs['download.default_directory'] = download_dir
-        chrome_options.add_experimental_option('prefs', prefs)
-        if headless:
-            chrome_options.add_argument('headless')
-        if use_seleniumrequests:
-            driver_class = seleniumrequests.Chrome
+        if not firefox:
+            self.chromedriver_bin = chromedriver_bin
+            chrome_options = webdriver.ChromeOptions()
+            chrome_options.add_argument("--no-sandbox")
+            service_args = []
+            chrome_options.add_experimental_option('excludeSwitches', [
+                'enable-automation',
+                'load-extension',
+                'load-component-extension',
+                'ignore-certificate-errors',
+                'test-type',
+            ])
+            if user_agent:
+                # Chrome/78.0.3904.70 Safari/537.36
+                chrome_options.add_argument("user-agent={}".format(user_agent))
+
+            if profile_dir is not None:
+                chrome_options.add_argument('user-data-dir=%s' % profile_dir)
+                if not os.path.exists(profile_dir):
+                    os.makedirs(profile_dir)
+            prefs = {}
+            prefs['plugins.plugins_disabled'] = [
+                'Chrome PDF Viewer', 'Chromium PDF Viewer'
+            ]
+            prefs['plugins.always_open_pdf_externally'] = True
+            if download_dir is not None:
+                prefs['download.default_directory'] = download_dir
+            chrome_options.add_experimental_option('prefs', prefs)
+            if headless:
+                chrome_options.add_argument('headless')
+            if use_seleniumrequests:
+                driver_class = seleniumrequests.Chrome
+            else:
+                driver_class = webdriver.Chrome
+            self.driver = driver_class(
+                executable_path=self.chromedriver_bin,
+                chrome_options=chrome_options,
+                service_args=service_args,
+            )
+            print(' --connect=%s --session-id=%s' %
+                  (self.driver.command_executor._url, self.driver.session_id))
+            signal.signal(signal.SIGINT, original_sigint_handler)
```

Comment:
I think this should be done for firefox as well.

Reply:
Done.